nixos/modules/programs/spacefm.nix

```diff
@@ -0,0 +1,41 @@
+# Global configuration for spacefm.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.spacefm;
+
+in
+{
+  ###### interface
+
+  options = {
+
+    programs.spacefm = {
+
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExample ''{
```

Comment:
Okay, I think you'll find that importing stuff into your main config isn't such a great strategy long term, especially if you're developing several features concurrently, but whatever floats your boat.